flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -1984,6 +1988,15 @@ IntrinsicLibrary::genChar(mlir::Type type,
   return fir::CharBoxValue{cast, len};
 }
 
+// COMMAND_ARGUMENT_COUNT
+fir::ExtendedValue
+IntrinsicLibrary::genCommandArgumentCount(mlir::Type resultType,
+                                          llvm::ArrayRef<fir::ExtendedValue> args) {
+  assert(args.size() == 0);
+  assert(resultType != builder.getDefaultIntegerType() && "result type is not default integer kind type");
```

Comment:
I think you may have changed the `==` to `!=` in the assert by mistake here (test now fails).

Reply:
Whoops. Thanks for spotting. Fixed now.